Cargo.toml

```diff
@@ -38,3 +38,32 @@ edition = "2021"
 debug = 0
 panic = 'unwind'
 incremental = false
+
+[lints.clippy]
+all = "warn"
+pedantic = "warn"
+struct_excessive_bools = "allow"
+trivially_copy_pass_by_ref = "allow"
+must_use_candidate = "allow"
+match_bool = "allow"
+no_effect_underscore_binding = "allow"
+module_name_repetitions = "allow"
+
+[lints.rust]
+meta_variable_misuse = "warn"
```

Comment:
can we add job which would consider every warning to be error instead? Just to make sure we won't accumulate warnings over time again.

Reply:
I thought we already touched down on this.

This can be controlled through a compiler flag which I plan on adding in a future PR. If I add it now, other branches (including this one) will fail in CI because there still are lints around.